src/App.tsx

```diff
@@ -8,7 +8,40 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+function shuffleTodos(array: Todo[]) {
+  const result = [...array];
+
+  for (let i = array.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1));
+
+    [result[i], result[j]] = [result[j], result[i]];
+  }
+
+  return result;
+}
+
 export const App: React.FC = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [loaded, setLoaded] = useState(false);
+  const [todo, setTodo] = useState<Todo>();
+  const [selectedTodo, setSelectedTodo] = useState(false);
+  const [filterType, setFilterType] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setInitialTodos(data);
+        setLoaded(true);
+      });
+  }, []);
+
+  const todos = useMemo(() => initialTodos.filter(currentTodo => (filterType === 'all' || (filterType === 'active') === (!currentTodo.completed))
+    && currentTodo.title.match(new RegExp(query, 'i'))), [filterType, query, initialTodos]);
```

Comment:
It is hard to read so long code

```suggestion
  const todos = useMemo(() => initialTodos.filter(currentTodo => 
  	(filterType === 'all' || (filterType === 'active') === (!currentTodo.completed))
  	&& currentTodo.title.match(new RegExp(query, 'i'))), [filterType, query, initialTodos]);
```